contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -71,37 +75,27 @@ contract FraudProofFacet is StateChannelCommon {
         return signer1; 
     }
 
-    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
+    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal pure returns (address) {
         BlockEmptyProof memory blockEmptyProof = abi.decode(encodedProof, (BlockEmptyProof));
-        StateSnapshot memory latestStateSnapshot = blockEmptyProof.latestStateSnapshot;
-        bytes32 previousStateSnapshotHash = blockEmptyProof.previousStateSnapshotHash;
-        bytes memory previousStateMachineState = blockEmptyProof.previousStateMachineState;
         Block memory fraudBlock = abi.decode(blockEmptyProof.emptyBlock.encodedBlock, (Block));
+        Block memory previousBlock = abi.decode(blockEmptyProof.previousBlock.encodedBlock, (Block));
 
         if(fraudProofVerificationContext.channelId != fraudBlock.transaction.header.channelId) {
             revert ErrorNotSameChannelId();
         }
 
+        if(fraudBlock.previousBlockHash != keccak256(abi.encode(previousBlock))){
+            revert ErrorLinkingPreviousBlock();
```

Comment:
Also check this after the blockHeight == 0 case and don't encode since you already have the encoded previous block in the proof, just reuse it